src/promptflow/promptflow/_sdk/_service/apis/flow.py

```diff
@@ -1,68 +1,170 @@
+    @api.doc(description="Flow test")
+    @api.expect(flow_test_model)
     def post(self):
-        args = flow_test_parser.parse_args()
-        if Configuration.get_instance().is_internal_features_enabled() and args.experiment:
+        flow = api.payload["flow"]
+        inputs = api.payload.get("inputs", None)
+        environment_variables = api.payload.get("environment_variables", None)
+        variant = api.payload.get("variant", None)
+        node = api.payload.get("node", None)
+        experiment = api.payload.get("experiment", None)
+        output_path = api.payload.get("detail", None)
+        if Configuration.get_instance().is_internal_features_enabled() and experiment:
             result = get_client_from_request().flows.test(
-                flow=args.flow,
-                inputs=args.inputs,
-                environment_variables=args.environment_variables,
-                variant=args.variant,
-                node=args.node,
-                experiment=args.experiment,
-                output_path=args.detail,
+                flow=flow,
+                inputs=inputs,
+                environment_variables=environment_variables,
+                variant=variant,
+                node=node,
+                experiment=experiment,
+                output_path=output_path,
             )
         else:
             result = get_client_from_request().flows.test(
-                flow=args.flow,
-                inputs=args.inputs,
-                environment_variables=args.environment_variables,
-                variant=args.variant,
-                node=args.node,
+                flow=flow,
+                inputs=inputs,
+                environment_variables=environment_variables,
+                variant=variant,
+                node=node,
                 allow_generator_output=False,
                 stream_output=False,
                 dump_test_result=True,
-                output_path=args.detail,
+                output_path=output_path,
             )
         return result
 
 
 @api.route("/get")
 class FlowGet(Resource):
-    @api.response(code=200, description="Get flow snapshot", model=dict_field)
-    @api.doc(parser=flow_test_parser, description="Get flow snapshot")
+    @api.response(code=200, description="Return flow yaml as json", model=dict_field)
+    @api.doc(description="Return flow yaml as json")
     def get(self):
-
-        args = flow_test_parser.parse_args()
-        source_path = Path(args.flow)
-        flow_path = resolve_flow_path(source_path)
+        flow_path = api.payload["flow"]
+        flow_path = resolve_flow_path(Path(flow_path))
         flow_info = load_yaml(flow_path)
         return flow_info
+
+
+@api.route("/ux_inputs")
+class FlowUxInputs(Resource):
+    @api.response(code=200, description="Get the file content of file UX_INPUTS_JSON", model=dict_field)
+    @api.doc(description="Get the file content of file UX_INPUTS_JSON")
+    def get(self):
```

Comment:
I think this is per flow based, how do we represent that?

Reply:
updated